linera-base/src/abi.rs

```diff
@@ -70,3 +72,24 @@ where
     type Query = <<A as WithServiceAbi>::Abi as ServiceAbi>::Query;
     type QueryResponse = <<A as WithServiceAbi>::Abi as ServiceAbi>::QueryResponse;
 }
+
+/// The type of a query for an oracle.
+pub trait OracleQuery: Send + Sync + Debug + 'static {
+    /// The oracle's ID.
+    const ID: OracleId;
```

Comment:
Mmm we haven't used magic numbers so far in ABIs.

Reply:
And I would prefer strings.